src/blocks/toggle.rs

```diff
@@ -39,10 +41,33 @@ pub struct ToggleConfig {
     /// Shell Command to determine toggle state. <br/>Empty output => off. Any output => on.
     pub command_state: String,
 
+    /// Icon ID when toggled on (default is "toggle_on")
+    #[serde(default = "ToggleConfig::default_icon_on")]
+    pub icon_on: String,
+
+    /// Icon ID when toggled off (default is "toggle_off")
+    #[serde(default = "ToggleConfig::default_icon_off")]
+    pub icon_off: String,
+
     /// Text to display in i3bar for this block
+    #[serde(default = "ToggleConfig::default_text")]
     pub text: String,
```

Comment:
Would it make sense to move to using an `Option<String>` instead? Serde should handle things correctly at the TOML level if so.

Reply:
I was looking for a way to do that! Thanks for the pointer. I'll play around with that.

Reply:
@atheriel PTAL at https://github.com/greshake/i3status-rust/pull/177/commits/49768cc2f001123017376c2049829499e727e97d, feels a bit better but not sure about the `with_content` builder.